### PR TITLE
Include all non-default global python kernel specs

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #6389

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [x] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
I want to be a little careful with this change (per Teams discussion). Per the previous function comment it looks pretty clear to me that the initial change was to exclude these kernels, and not just the default ones. Right now with the kernels that I register using --user like so:
https://ipython.readthedocs.io/en/stable/install/kernel_install.html#kernels-for-different-environments
I can create kernelspecs that all point at the originating interpreter path. And since the full path is specified they don't look ambiguous as the kernel finder code correctly matches them up with the right interpreter (conda kernelspecs that I created match up with the conda env that created them. Same with venvs).

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Test issues. Will hold off and investigate on this.

**Review by DavidKutu — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
We chatted on teams quick. I think there is more going on here. Don't want to take this fix for release. Will update #6389 with more info.

File: src/client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder.ts [hunk 1]
```diff
@@ -107,8 +107,12 @@ export class LocalPythonAndRelatedNonPythonKernelSpecFinder extends LocalKernelS
...
         ]);
 
-        const globalPythonKernelSpecsRegisteredByUs = globalKernelSpecs.filter((item) =>
-            isKernelRegisteredByUs(item.kernelSpec)
+        // Only return global location kernelspecs if they are the old ones registered by us, or if they
+        // do not match the default kernel spec name.
+        const nonDefaultGlobalPythonKernelSpecs = globalKernelSpecs.filter(
+            (item) =>
+                isKernelRegisteredByUs(item.kernelSpec) ||
+                !item.kernelSpec.name.toLowerCase().match(isDefaultPythonKernelSpecName)
```
Comment (IanMatthewHuff) [inline src/client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder.ts:115]: Note, this isn't right (we are already removing default below). More discussion in the linked issue.

**Review by raghavthind2005 — APPROVED** [review]

**Pull request closed without merging**